awot/io/read_sonde.py

```diff
@@ -137,27 +137,198 @@ def read_dropsonde_data(filePath, split_file=True):
+        
+        output:
+        
+        #float
+        '''
+    f = open(filename, 'r')
+    nlines = len(f.readlines())
+    f.close()
+    return nlines
+
+
+
+
+
+def _get_header(f):
+    '''
+        data_type : Number of header lines
+        project : NASA AMES FFI format number
+        site : Originator/PI Name
+        site_type : Name of organization
+        site_id : Instrument/platform name
+        lon0_dms : Project/mission name
+        lat0_dms : Current volume number (almost always 1)
+        lat0_dec : Number of volumes for data (almost always 1)
+        alt : YYYY MM DD UTC begin date
+        datetime0 : Reduction/revision UTC date
+        launch_reference : Interval between successive values (data rate)
+        sonde_id : Name/Description of DX variable above
+        op_comments : Number of primary variables in file
+        proc_comments : Scaling factor for each variable column
+        varnames : Missing value for each variable column
+        varunits : Name of first variable
+        '''
+    hdr = {}
+    #List index out of range? See bototm of notebook
+    hdr['data_type'] = f.readline().rstrip('\n').split(":")[1].strip()
+    
+    hdr['project'] = f.readline().rstrip('\n').split(":")[1].strip()
+    hdr['site'], hdr['site_type'], hdr['site_id']  = f.readline().rstrip('\n').split(":")[1].split()
+    lonlatalt= f.readline().rstrip('\n').split(":")[1].strip()
+    hdr['lon0_dms'], hdr['lat0_dms'], hdr['lon0_dec'], hdr['lat0_dec'], hdr['alt'] = lonlatalt.split(',')
+    # Turn some values to float
+    hdr['lon0_dec'] = float(hdr['lon0_dec'])
+    hdr['lat0_dec'] = float(hdr['lat0_dec'])
+    hdr['alt'] = float(hdr['alt'])
+    release_time = f.readline().rstrip('\n').split(":")
+    yyyy, mm, dd, hh = release_time[1].strip().split(',')
+    nn = release_time[2].strip()
+    ss = release_time[2].strip()
+    dstring = "%s-%s-%sT%s:%s:%s" % (int(yyyy), int(mm), int(dd), int(hh), int(nn), int(ss))
+    hdr['datetime0'] = datetime(int(yyyy), int(mm), int(dd), int(hh), int(nn), int(ss))
+    hdr['launch_reference'] = f.readline().rstrip('\n').split(":")[1].strip() # Not complete - Throwaway
+    hdr['sonde_id'] = f.readline().rstrip('\n').split(":")[1].strip()
+    hdr['op_comments'] = f.readline().rstrip('\n').split(":")[1].strip()
+    hdr['proc_comments'] = f.readline().rstrip('\n').split(":")[1].strip()
+    f.readline() # Empty line '/'
+    f.readline() # Empty line '/'
+    f.readline() # This is the Nominal Release Time - Repeate datetime0
+    hdr['varnames'] = f.readline().rstrip('\n').strip().split()
+    hdr['varunits'] = f.readline().rstrip('\n').strip().split()
+    f.readline() # Divider line
+    
+    return hdr
+
+
+def read_cls_file(filename):
+    '''
+    #Method designed to read and seperate CLS Files from Noaa p3 Dropsonde data.
+    
+    Input:
+    #FilePath
+    
+    Output:
+    Dictionary containing file instances as sorted by date of launch and variable recorded.
+    

```

Comment:
There is a duplicate ```read_cls_file``` method. Probably should combine the docstring from the empty method into working method.